### DISCUSSION TITLE
Unexpected Token On Client Fetch

### DISCUSSION THREAD

**Discussion by SkelatorIndy** [nextauthjs/next-auth#2001] [Help] [answered]
I have setup NextAuth with my app, and locally, everything works as expected. However, after deploying to Vercel, there are errors with NextAuth, specifically an "Unexpected token < in JSON at position 0" when NextAuth makes the "/api/auth/session" request. Locally, when navigating to that URL, it returns an object, as it should. When I go to that URL with my production domain, it returns an empty page of my app, not the object it should. I can't seem to find any discussions that have this issue, and I'm not sure why it's happening since I don't get any other info from the console other than the unexpected token one. I can try to provide any info needed to figure out why this is happening, just ask for it.

**Comment by balazsorban44** [comment, marked as answer]
You will need to provide a reproduction if you expect any help. Check out our official example: https://github.com/nextauthjs/next-auth-example

**Comment by lveillard** [comment]
@SkelatorIndy were you able to find the issue? I have exactly the same problem

**Reply by balazsorban44** [reply]
Have a reproduction? Check your Vercel logs to further investigate as well.

**Reply by lveillard** [reply]
Hello there! Yep my bad, i checked the logs and it was clear, the .env variables where not correctly set up. In my case using render.com but yes, to anyone with this kind of issue just go to the logs.

Also don't forget to setup NEXTAUTH_URL in the environemment variables

**Reply by balazsorban44** [reply]
You don't actually need to set `NEXTAUTH_URL` on Vercel.
